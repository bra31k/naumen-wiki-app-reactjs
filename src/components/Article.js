import React from 'react';
import PropTypes from "prop-types"
import "../style/index.css"

const href = 'https://en.wikipedia.org/wiki/'

export const Article = ({title, snippet}) =>  {
        return (
            <div className="article">
                <h3><a href={href + title} target="_blank" rel="noopener noreferrer">{title}</a></h3>
                <p dangerouslySetInnerHTML={{__html: snippet}}/>
            </div>
        );
}


Article.propTypes = {
    title: PropTypes.string,
    snippet: PropTypes.string,
};

export default Article;
