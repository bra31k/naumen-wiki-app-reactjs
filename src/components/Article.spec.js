import React from 'react'
import { Article } from './Article'
import { shallow } from 'enzyme'

describe('Article component', () => {
    const props = {
        title: 'hello',
        snippet: 'world'
    }

    describe('Article initial', () => {
        const article = shallow(<Article {...props} />)

        it('renders properly', () => {
            expect(article).toMatchSnapshot()
        })
    })
})