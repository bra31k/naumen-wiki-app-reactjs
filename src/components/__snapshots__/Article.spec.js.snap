// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component Article initial renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article
    snippet="world"
    title="hello"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          <a
            href="https://en.wikipedia.org/wiki/hello"
            rel="noopener noreferrer"
            target="_blank"
          >
            hello
          </a>
        </h3>,
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "world",
            }
          }
        />,
      ],
      "className": "article",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="https://en.wikipedia.org/wiki/hello"
            rel="noopener noreferrer"
            target="_blank"
          >
            hello
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "hello",
            "href": "https://en.wikipedia.org/wiki/hello",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "hello",
          "type": "a",
        },
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "dangerouslySetInnerHTML": Object {
            "__html": "world",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            <a
              href="https://en.wikipedia.org/wiki/hello"
              rel="noopener noreferrer"
              target="_blank"
            >
              hello
            </a>
          </h3>,
          <p
            dangerouslySetInnerHTML={
              Object {
                "__html": "world",
              }
            }
          />,
        ],
        "className": "article",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="https://en.wikipedia.org/wiki/hello"
              rel="noopener noreferrer"
              target="_blank"
            >
              hello
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "hello",
              "href": "https://en.wikipedia.org/wiki/hello",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "hello",
            "type": "a",
          },
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "world",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
